applications/FluidDynamicsApplication/custom_processes/mass_conservation_check_process.cpp

```diff
@@ -0,0 +1,265 @@
+
+// Application includes
+#include "mass_conservation_check_process.h"
+#include "fluid_dynamics_application_variables.h"
+
+#include "custom_elements/two_fluid_navier_stokes.h"
+#include "custom_utilities/two_fluid_navier_stokes_data.h"
+#include "modified_shape_functions/tetrahedra_3d_4_modified_shape_functions.h"
+#include "modified_shape_functions/triangle_2d_3_modified_shape_functions.h"
+
+namespace Kratos
+{
+
+/* Public functions *******************************************************/
+
+MassConservationCheckProcess::MassConservationCheckProcess(
+        ModelPart& rModelPart, 
+        const bool IsSwitchedOn,
+        const int MassComputationFreq,
+        const bool CompareToInitial, 
+        const bool WriteToLogFile)
+    : Process(), mrModelPart(rModelPart) {
+
+    mIsSwitchedOn = IsSwitchedOn;
+    mMassComputationFreq = MassComputationFreq;
+    mCompareToInitial = CompareToInitial;
+    mWriteToLogFile = WriteToLogFile;
+}
+
+MassConservationCheckProcess::MassConservationCheckProcess(
+    ModelPart& rModelPart,
+    Parameters& rParameters)
+    : Process(), mrModelPart(rModelPart) {
+
+    Parameters default_parameters( R"(
+    {
+        "model_part_name"                        : "default_model_part_name",
+        "is_switched_on"                         : true,
+        "mass_computation_frequency"             : 20,
+        "compare_to_initial_values"              : true,
+        "write_to_log_file"                      : true
+    }  )" );
+
+    rParameters.ValidateAndAssignDefaults(default_parameters);
+
+    mIsSwitchedOn = rParameters["is_switched_on"].GetBool();
+    mMassComputationFreq = rParameters["mass_computation_frequency"].GetInt();
+    mCompareToInitial = rParameters["compare_to_initial_values"].GetBool();
+    mWriteToLogFile = rParameters["write_to_log_file"].GetBool();
+}
+
+// void DistanceModificationProcess::ExecuteInitialize() {
+
+//     KRATOS_TRY;
+
+//     // Continuous distance field required variables check
+//     if (mContinuousDistance){
+//         const auto& r_node = *mrModelPart.NodesBegin();
+//         KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(NODAL_H, r_node);
+//         KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DISTANCE, r_node);
+//     }
+
+//     KRATOS_CATCH("");
+// }
+
+void MassConservationCheckProcess::ExecuteFinalizeSolutionStep()
+{
+  
+    if ( this->mIsSwitchedOn ){
+            double negVol = 0.0;
+            double posVol = 0.0;
+
+        if ( mCompareToInitial && mCounter == 1){
+            // getting initial values and storing as reference
+            this->ComputeVolumesOfFluids( posVol, negVol );
+            mInitialPositiveVolume = posVol;
+            mInitialNegativeVolume = negVol;
+        }
+
+        if ( mCounter % this->mMassComputationFreq == 0){
+            // writing an output at a given frequncy
+            this->ComputeVolumesOfFluids( posVol, negVol );
+            std::cout << " --- Volume Checking Process --- " << std::endl;
```

Comment:
Do you really want all this spurious info? Maybe in FullDebug mode or with a more severe echo level? In any case you must use the Kratos logger instead.

Reply:
It was removed from the Cpp and included in the python file. Also the file handling was moved to python as recommended by Phillipp.